dawn/src/dawn/CodeGen/CXXNaive-ico/CXXNaiveCodeGen.cpp

```diff
@@ -247,28 +246,56 @@ void CXXNaiveIcoCodeGen::generateStencilWrapperCtr(
     }
     initCtr += ")";
     StencilWrapperConstructor.addInit(initCtr);
-  }
 
-  if(metadata.hasAccessesOfType<iir::FieldAccessType::InterStencilTemporary,
-                                iir::FieldAccessType::StencilTemporary>()) {
-    for(auto accessID : metadata.getAccessesOfType<iir::FieldAccessType::InterStencilTemporary,
-                                                   iir::FieldAccessType::StencilTemporary>()) {
-
-      if(metadata.isMultiVersionedField(accessID)) {
-        int originalAccessID = metadata.getOriginalVersionOfAccessID(accessID);
-        StencilWrapperConstructor.addInit("m_" + metadata.getNameFromAccessID(accessID) +
-                                          "(allocateFieldLike(LibTag{}, " +
-                                          metadata.getNameFromAccessID(originalAccessID) + "))");
-      } else {
-        StencilWrapperConstructor.addInit("m_" + metadata.getNameFromAccessID(accessID) +
-                                          "(allocateFieldLike(LibTag{}, " +
-                                          metadata.getNameFromAccessID(accessID) + "))");
+    if(metadata.hasAccessesOfType<iir::FieldAccessType::InterStencilTemporary,
+                                  iir::FieldAccessType::StencilTemporary>()) {
+      for(auto accessID : metadata.getAccessesOfType<iir::FieldAccessType::InterStencilTemporary,
+                                                     iir::FieldAccessType::StencilTemporary>()) {
+
+        if(metadata.isMultiVersionedField(accessID)) {
+          int originalAccessID = metadata.getOriginalVersionOfAccessID(accessID);
+          StencilWrapperConstructor.addInit("m_" + metadata.getNameFromAccessID(accessID) +
+                                            "(allocateFieldLike(LibTag{}, " +
+                                            metadata.getNameFromAccessID(originalAccessID) + "))");
```

Comment:
Now that there is an implementation (else block below) to handle allocation of a temporary in general (without the need to replicate the structure of another field), the same can be used to do the allocation for this case.
In other words, the code below can already handle both cases and you don't need `allocateFieldLike(...)` anymore.